### PR TITLE
Fix armor attack class for modify attribute effects

### PR DESCRIPTION
**Pull request by mardaravicius**
these 2 effects combine variable and armor/attack type into 1 var `variable`. there the least significant bits are used for variable and the most significant bits for the atk/armor class. an int is split into 2 bytes for each it seems.

### CODE REVIEW

File: AoE2ScenarioParser/objects/support/new_effect.py [hunk 1]
```diff
@@ -1565,16 +1566,25 @@ def modify_attribute_by_variable(
             object_attributes: int | None = None,
             variable: int | None = None,
             message: str | None = None,
+            armour_attack_class: int | None = None,
     ):
-        return self._trigger_ref._add_effect(
+        if (armour_attack_class is not None and object_attributes != ObjectAttribute.ATTACK and
```
Comment (mardaravicius) [inline AoE2ScenarioParser/objects/support/new_effect.py:1571]: might need to check for -1 here as well

File: AoE2ScenarioParser/objects/support/new_effect.py [hunk 2]
```diff
@@ -1565,16 +1566,25 @@ def modify_attribute_by_variable(
...
             object_list_unit_id=object_list_unit_id,
             source_player=source_player,
             operation=operation,
             object_attributes=object_attributes,
             variable=variable,
-            message=message
+            message=message,
         )
+        if armour_attack_class is not None and armour_attack_class > -1:
+            combined_var = effect._aa_to_quantity(variable, armour_attack_class)
```
Comment (mardaravicius) [inline AoE2ScenarioParser/objects/support/new_effect.py:1584]: might need to refac tor so this method is more easily accessible here

Comment (mardaravicius) [reply AoE2ScenarioParser/objects/support/new_effect.py:1584]: also it is more intuitive that the least significant bits would be the 2nd argument of the function. currently its the reverse

**Comment by mardaravicius** [comment]
when using it with this script it seems to work:
```python
from AoE2ScenarioParser.datasets.trigger_lists import Operation, ObjectAttribute
from AoE2ScenarioParser.datasets.units import UnitInfo
from AoE2ScenarioParser.scenarios.aoe2_de_scenario import AoE2DEScenario


def main():
    scenario_dir = "<scenario_dir>"
    scenario = AoE2DEScenario.from_file(scenario_dir)
    trigger = scenario.trigger_manager.add_trigger("test")

    trigger.new_effect.change_variable(variable=10, quantity=10)
    trigger.new_effect.modify_attribute_by_variable(
        object_list_unit_id=UnitInfo.TEUTONIC_KNIGHT.ID,
        source_player=1,
        operation=Operation.ADD,
        object_attributes=ObjectAttribute.ATTACK,
        variable=10,
        message="Test msg 10",
        armour_attack_class=4,
    )
    trigger.new_effect.modify_variable_by_attribute(
        object_list_unit_id=UnitInfo.TEUTONIC_KNIGHT.ID,
        source_player=1,
        operation=Operation.SET,
        object_attributes=ObjectAttribute.ARMOR,
        variable=11,
        message="Test msg 11",
        armour_attack_class=4,
    )
    trigger.new_effect.script_call(message="void t(){xsChatData(\"var was: %d \",xsTriggerVariable(11));}")


    scenario.write_to_file(scenario_dir)


if __name__ == "__main__":
    main()
```

**Comment by KSneijders** [comment]
Hey, thanks for the script and PR. Unfortunately this is not exactly how I'd like it to be used.
Using this approach, you can no longer use or edit the `variable` attribute on an `Effect` instance.

I'll be pushing a commit which modifies this PR quite heavily. It uses the same logic and systems (although also renewed) used for the other armour attack effects.

I won't be merging today anymore, so feel free to check it out if you like. Any remarks or questions are always welcome!

**Comment by KSneijders** [comment]
@mardaravicius Hey, thank you so much for the Pull Request. Unfortunately this is not exactly how I'd like it to work. Using this approach, you can no longer use or edit the `variable` attribute on an `Effect` instance.

I'll be pushing a commit which modifies this PR quite heavily. It uses the same logic and systems (although also renewed) used for the other armour attack effects.

I won't be merging today anymore, so feel free to check it out if you like. Any remarks or questions are always welcome!

**Pull request merged**